src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -32,7 +32,7 @@ private void InitSort(CultureInfo culture)
             }
             else
             {
-                _isAsciiEqualityOrdinal = (_sortName == "en-US" || _sortName == "");
+                _isAsciiEqualityOrdinal = _sortName == "" || (_sortName.Length >= 2 && _sortName[0] == 'e' && _sortName[1] == 'n');
```

Comment:
Are all cultures either two characters or more than two followed by a dash?

Reply:
There is one culture called just "en", hence the lack of dash check here.

<details>

```cs
foreach(var culture in CultureInfo.GetCultures(CultureTypes.AllCultures).Where(c => c.Name.StartsWith("en")))
{
    Console.WriteLine(culture.Name);
}
```

```log
en
en-001
en-029
en-150
en-AG
en-AI
en-AS
en-AT
en-AU
en-BB
en-BE
en-BI
en-BM
en-BS
en-BW
en-BZ
en-CA
en-CC
en-CH
en-CK
en-CM
en-CX
en-CY
en-DE
en-DK
en-DM
en-ER
en-FI
en-FJ
en-FK
en-FM
en-GB
en-GD
en-GG
en-GH
en-GI
en-GM
en-GU
en-GY
en-HK
en-ID
en-IE
en-IL
en-IM
en-IN
en-IO
en-JE
en-JM
en-KE
en-KI
en-KN
en-KY
en-LC
en-LR
en-LS
en-MG
en-MH
en-MO
en-MP
en-MS
en-MT
en-MU
en-MW
en-MY
en-NA
en-NF
en-NG
en-NL
en-NR
en-NU
en-NZ
en-PG
en-PH
en-PK
en-PN
en-PR
en-PW
en-RW
en-SB
en-SC
en-SD
en-SE
en-SG
en-SH
en-SI
en-SL
en-SS
en-SX
en-SZ
en-TC
en-TK
en-TO
en-TT
en-TV
en-TZ
en-UG
en-UM
en-US
en-VC
en-VG
en-VI
en-VU
en-WS
en-ZA
en-ZM
en-ZW
```

</details>

Reply:
> There is one culture called just "en", hence the lack of dash check here.

I understand. I was wondering if it's possible there could be a culture e.g. ene-ab, with three chars before a dash.

Reply:
Right, we need to check ```en-``` and not only ```en```